systems/trajectory_optimization/test/cost_constraint_approximation_test.cc

```diff
@@ -0,0 +1,114 @@
+#include <Eigen/Dense>
+#include <memory>
+#include <utility>
+#include <gtest/gtest.h>
+
+#include "drake/solvers/mathematical_program.h"
+#include "drake/solvers/snopt_solver.h"
+#include "drake/solvers/solve.h"
+#include "systems/trajectory_optimization/dircon_util.h"
+
+namespace dairlib {
+namespace systems {
+namespace {
+
+using drake::solvers::MathematicalProgram;
+using std::cout;
+using std::endl;
+using Eigen::Vector3d;
+using Eigen::VectorXd;
+using Eigen::MatrixXd;
+
+class CostConstraintApproximationTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    // An example
+    H_o_ <<  1.36075, 0.354964,
+         0.354964,  1.19376;
+    b_o_ = VectorXd::Ones(2);
+    c_o_ << 0;
+    A_o_ <<  0.411647, -0.164777,
+         -0.302449, 0.26823;
+    lb_o_ = 0.5 * VectorXd::Ones(2);
+    ub_o_ = VectorXd::Ones(2);
+
+    // solve optimization problem
+    MathematicalProgram quadprog;
+    auto w = quadprog.NewContinuousVariables(2, "w");
+    // intentionally split the following constraint into two constraint bindings
+    quadprog.AddLinearConstraint(A_o_.row(0),
+                                 lb_o_(0),
+                                 ub_o_(0),
+                                 w);
+    quadprog.AddLinearConstraint(A_o_.row(1),
+                                 lb_o_(1),
+                                 ub_o_(1),
+                                 w);
+    // Adds a cost term of the form 0.5*x'*H_o*x + b_o'x + c
+    // intentionally split the following cost into two cost bindings
+    quadprog.AddQuadraticCost(H_o_ * 1 / 3, b_o_ * 1 / 3, w);
+    quadprog.AddQuadraticCost(H_o_ * 2 / 3, b_o_ * 2 / 3, w);
+    const auto result = Solve(quadprog);
+    w_sol_ = result.GetSolution(quadprog.decision_variables());
+
+    // Approximation
+    dairlib::systems::trajectory_optimization::linearizeConstraints(
```

Comment:
The functions you're testing should probably be in the test itself, not the setup. Setup is largely useful for any shared constructors/problem setup that will be used across multiple tests without changing.

Reply:
Also, there's nothing special about using `w_sol_`, any `w` could be used for this test.

Reply:
> The functions you're testing should probably be in the test itself, not the setup. Setup is largely useful for any shared constructors/problem setup that will be used across multiple tests without changing.

Got it!

Reply:
> Also, there's nothing special about using `w_sol_`, any `w` could be used for this test.

I used w_sol because w is the decision variable